### PR TITLE
Create AlgorithmRunner with no Qt

### PR DESCRIPTION
**Pull request by robertapplin**
### Description of work
This PR renames the old `AlgorithmRunner` class to `QtAlgorithmRunner` because it is a QObject and uses qt signals. The PR also creates a new class called `AlgorithmRunner` which does the same as `QtAlgorithmRunner` but without qt signals. The plan is to use the new  `AlgorithmRunner` class in the Indirect/Inelastic interfaces.

*There is no associated issue.*

### To test:
Do some testing of the Reflectometry interface
I have already done testing of the Indirect interfaces
Check the new `AlgorithmRunnerTest` passes

*This does not require release notes* because **it is an internal change**

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Review by rbauststfc — COMMENTED** [review]
I've done a thorough test of the Reflectometry GUI, checking that we can execute batches, pause them during execution and then start them again. The progress bar updates correctly and GUI controls enable and disable correctly in response to batch reductions starting, stopping and completing. So everything looks good from the Reflectometry side. The code changes also all look good.

@robertapplin, you mention that you've tested the Indirect interfaces yourself. I just wanted to check, as this is your PR, are you happy that we don't need to have someone that hasn't worked on the changes do some testing of those interfaces?

**Comment by robertapplin** [comment]
> @robertapplin, you mention that you've tested the Indirect interfaces yourself. I just wanted to check, as this is your PR, are you happy that we don't need to have someone that hasn't worked on the changes do some testing of those interfaces?

Thanks for testing this so quick! I am happy that this hasn't broken anything in those interfaces, so I think we should be ok (famous last words!)

**Review by rbauststfc — APPROVED** [review]
Thanks Rob - in which case, I think this is good to go in 👍

File: qt/widgets/common/src/AlgorithmRunner.cpp [hunk 1]
```diff
@@ -1,120 +1,47 @@
...
-  m_asyncAlg = alg;
-  m_asyncResult = new Poco::ActiveResult<bool>(m_asyncAlg->executeAsync());
+void AlgorithmRunner::execute(std::deque<IConfiguredAlgorithm_sptr> algorithmQueue) {
+  m_jobRunner->setAlgorithmQueue(algorithmQueue);
+  m_jobRunner->executeAlgorithmQueue();
 }
 
-/// Get back a pointer to the running algorithm
-Mantid::API::IAlgorithm_sptr AlgorithmRunner::getAlgorithm() const { return m_asyncAlg; }
+void AlgorithmRunner::notifyBatchComplete(bool error) { m_subscriber->notifyBatchComplete(m_lastAlgorithm, error); };
```
Comment (thomashampson) [inline qt/widgets/common/src/AlgorithmRunner.cpp:29]: extra semi-colon

Comment (robertapplin) [reply qt/widgets/common/src/AlgorithmRunner.cpp:29]: Removed in last commit of this https://github.com/mantidproject/mantid/pull/37255

**Pull request merged**